src/apps/tools/OswAppCalculator.cpp

```diff
@@ -0,0 +1,448 @@
+/*
+    a calculator for addition, subtraction, multiplication, division, square roots, powers, num1 * 10 ^ num2
+    with postitive and negativ numebers + generating random numbers
+    press btn2 + btn3 to set the cursor a position back
+*/
+
+#include "./apps/tools/OswAppCalculator.h"
+
+#include "config_defaults.h"  // Include the config here again to access the language definitions of it
+#include "gfx_util.h"
+#include "osw_app.h"
+#include "osw_hal.h"
+#include "osw_ui.h"
+
+#include <osw_config_keys.h>
+
+char pNum1 = '+';
+int num1[] = {0, 0, 0, 0, 0, 0};
+double Num1;
+String op = "+";
+char pNum2 = '+';
+int num2[] = {0, 0, 0, 0, 0, 0};
+double Num2;
+int8_t i = 0;
+uint8_t iNum = 0;
+double result;
+String resultS;
```

Comment:
Please refrain from using static / global variables inside your application! It wastes resources and slows down the boot time - PLEASE add them as member variables into your class.

Reply:
I removed nearly all, what was a big challenge, but I learned a lot of oop. Do global variables only slow down the app that they are used in, or are ther other disadvantages? By "add them as member variables into your class" did you mean I should move the variables into the appropriate
appropriate functions?

Reply:
oh, now if seen that I also shouldnt youse static variables. Can you give me an example what I should do instead? Or refer me to a tutorial. Im pretty lost ther.

Reply:
Do not worry, we are now on it: https://github.com/Open-Smartwatch/open-smartwatch-os/pull/294/files#r1043267117